ecc_enclave/enclave/enclave.cpp

```diff
@@ -67,14 +69,123 @@ int ecall_bind_tlcc(const sgx_report_t* report, const uint8_t* pubkey)
     return SGX_SUCCESS;
 }
 
-int invoke_enc(const char* args,
-    const char* pk,
+int gen_response(const char* txType,
+    uint8_t* response,
+    uint32_t* response_len_out,
+    sgx_ec256_signature_t* signature,
+    shim_ctx_ptr_t ctx)
+{
+    int ret;
+
+    // Note: below signature is verified in
+    // - ecc/crypto/ecdsa.go::Verify (for VSCC)
+    // - tlcc_enclave/enclave/ledger.cpp::int parse_endorser_transaction (for TLCC)
+
+    // create Hash <- H(txType in {"init", "invoke"} || encoded_args || result || read set || write
+    // set)
+    // TODO: we should encode the hash below in an unambiguous fashion (which is not true with
+    //    simple concatenation as done below!)
+    //    Probably easiest by prefixing each field by length in fixed-size format?
+    sgx_sha256_hash_t hash;
+    sgx_sha_state_handle_t sha_handle;
+    sgx_sha256_init(&sha_handle);
+    LOG_DEBUG("txType: %s\n", txType);
```

Comment:
`\n` leads to extra new lines which are not needed here. For all LOG_DEBUG statements no need for \b

Reply:
i think it got there from copy/paste from existing log statement. but eradicated now all \n throughout code.   Interestingly, i also never really noticed any empty lines in the logs, though.